tests/generator-Tests/Unit-Tests/CodeGeneratorExpectedResults/JavaInterop1/WriteClass.txt

```diff
@@ -260,6 +261,7 @@ public partial class MyClass  {
 			return __rm;
 		} finally {
 			JNIEnv.DeleteLocalRef (native_key);
+			global::System.GC.KeepAlive (key);
```

Comment:
***Does not need to be addressed for d16-8***: We don't need a `GC.KeepAlive()` for types which will be "deep marshaled".  This includes `System.String` and array types.  ***For master***, would we be able to *skip* this?

Reply:
Yes, we should be able to do that.

For the array types, would we never need the `KeepAlive`, or does it depend on the array element type?

ie: `int[]` versus `object[]`

Reply:
I think it *will* depend on the array element type, as if we have a `Java.Lang.Object[]`, then everything within the array needs to be kept alive, and `GC.KeepAlive(array)` will ensure that everything within the array is kept alive…

So maybe *all* array types *should* retain the `GC.KeepAlive()`…